payment/manual/renew-manual.go

```diff
@@ -0,0 +1,107 @@
+package manual
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+	"github.com/wopta/goworkspace/network"
+	plcRenew "github.com/wopta/goworkspace/policy/renew"
+	prd "github.com/wopta/goworkspace/product"
+	tr "github.com/wopta/goworkspace/transaction"
+	trRenew "github.com/wopta/goworkspace/transaction/renew"
+)
+
+func RenewManualPaymentFx(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	var (
+		err         error
+		payload     ManualPaymentPayload
+		policy      models.Policy
+		transaction *models.Transaction
+		mgaProduct  *models.Product
+		networkNode *models.NetworkNode
+	)
+
+	log.SetPrefix("[RenewManualPaymentFx] ")
+	defer func() {
+		if err != nil {
+			log.Printf("error: %s", err)
+		}
+		log.Printf("Handler end -----------------------------------------------")
+		log.SetPrefix("")
+	}()
+	log.SetPrefix("Handler start ----------------------------------------------")
+
+	err = json.NewDecoder(r.Body).Decode(&payload)
+	if err != nil {
+		return "", nil, err
+	}
+
+	methods := models.GetAllPaymentMethods()
+	isMethodAllowed := lib.SliceContains(methods, payload.PaymentMethod)
+
+	if !isMethodAllowed {
+		return "", nil, errors.New(errPaymentMethodNotAllowed)
+	}
+
+	transactionUid := chi.URLParam(r, "transactionUid")
+	transaction = trRenew.GetRenewTransactionByUid(transactionUid)
+	if transaction == nil {
+		return "", nil, errors.New("no renew transaction found")
+	}
+	if transaction.IsPay {
+		return "", nil, errors.New(errTransactionPaid)
+	}
+
+	policy, err = plcRenew.GetRenewPolicyByUid(transaction.PolicyUid)
+	if err != nil {
+		return "", nil, err
+	}
+
+	isFirstTransactionAnnuity := policy.StartDate.AddDate(policy.Annuity, 0, 0) != transaction.EffectiveDate
+	if !isFirstTransactionAnnuity {
+		return "", nil, errors.New("cannot pay transaction that is not the first")
+	}
```

Comment:
I'm not sure this is the business requirement but I agree that in this view we should be able to only act on the transaction that modifies also the draft, i.e. the first transaction of a given annuity. All other actions on transactions should be made in the centralized view of policy.